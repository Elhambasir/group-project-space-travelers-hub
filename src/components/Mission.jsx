import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchMissions } from '../redux/mission/missionSlice';

const Mission = () => {
  const dispatch = useDispatch();
  const missions = useSelector((store) => store.missions);
  useEffect(() => {
    if (!missions.length) dispatch(fetchMissions());
  }, [dispatch]); // eslint-disable-line
  return (
    <div>hello world</div>
  );
};

export default Mission;
