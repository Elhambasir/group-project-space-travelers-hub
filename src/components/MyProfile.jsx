import React from 'react';
import { useSelector } from 'react-redux';
import { Container, ListGroup } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';


function Mission() {
  const mission = useSelector((state) => state.missions);
  const myMission = mission.missions.filter((item) => item.reserved === true);
  if (myMission.length === 0) return (<Container>You have not joined any Mission.</Container>);
  return (
    <Container>
      <ListGroup as="ol" numbered>
        {myMission.map((item) => (
          <ListGroup.Item as="li" key={item.id}>{item.name}</ListGroup.Item>
        ))}
      </ListGroup>
    </Container>
  );
}

const Rocket = () => {
  const Rocket = useSelector((state) => state.rockets);
  const myMission = Rocket.RocketList.filter((item) => item.reserved === true);
  if (myMission.length === 0) return (<Container>You have not reserved any Rocket.</Container>);
  return (
    <Container>
      <ListGroup as="ol" numbered>
        {myMission.map((item) => (
          <ListGroup.Item as="li" key={item.id}>{item.name}</ListGroup.Item>
        ))}
      </ListGroup>
    </Container>
  );
};

const MyProfile = () => (
  <>
    <Mission />
    <Rocket />
  </>
);

export default MyProfile;
