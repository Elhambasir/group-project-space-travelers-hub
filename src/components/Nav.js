const Nav = () => <nav />;

export default Nav;
