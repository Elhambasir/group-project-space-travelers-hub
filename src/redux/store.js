// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import MissionReducer from './mission/missionSlice';

// eslint-disable-next-line import/prefer-default-export
export const store = configureStore({
  reducer: {
    missions: MissionReducer,
  },
});
