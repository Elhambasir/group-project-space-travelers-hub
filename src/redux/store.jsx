// eslint-disable-next-line import/no-extraneous-dependencies
import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './Rockets/rocketsSlice';
import MissionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: MissionReducer,
  },
});

export default store;
