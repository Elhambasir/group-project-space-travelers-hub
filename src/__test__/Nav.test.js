import RocketList from '../components/RocketsList';
import Mission from '../components/Mission';
import MyProfile from '../components/MyProfile';
import Nav from '../components/Nav';
import * as React from "react";
import { create } from "react-test-renderer";
import {
  MemoryRouter,
  Routes,
  Route,
} from "react-router-dom";
import { Provider } from 'react-redux';
import store from '../redux/store';

describe("My app", () => {
  it("renders correctly", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/"]}>
        <Provider store={store}>
        <Nav />
        <Routes path="/" element={<RocketList />}>
          <Route index element={<RocketList />} />
          <Route path="/MyProfile" element={<MyProfile />} />
          <Route path="/Mission" element={<Mission />} />
          <Route path="*" element={<h3>No Match</h3>} />
        </Routes>
        </Provider >
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});