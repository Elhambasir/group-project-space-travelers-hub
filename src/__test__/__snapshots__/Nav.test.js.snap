// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My app renders correctly 1`] = `
Array [
  <div
    className="d-flex flex-wrap align-items-center p-3 container"
  >
    <img
      alt="Logo"
      className="Logo"
      src="https://spacetravellerhubcapstone.netlify.app/static/media/planet.3f6fb2742f28651730c0.png"
    />
    <h3>
      Space Travelers' Hub
    </h3>
    <ul
      className="list-unstyled m-2 p-2 d-flex flex-wrap"
      style={
        Object {
          "flex": "2",
          "justifyContent": "flex-end",
        }
      }
    >
      <li
        className="m-2"
      >
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="m-2"
      >
        <a
          className="link"
          href="/Mission"
          onClick={[Function]}
        >
          Mission
        </a>
      </li>
      <li
        className="m-2"
      >
        <a
          className="link"
          href="/MyProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>,
  <div>
    Loading
  </div>,
]
`;
